pkg/clusteragent/custommetrics/provider.go

```diff
@@ -45,13 +45,61 @@ type datadogProvider struct {
 // NewDatadogProvider creates a Custom Metrics and External Metrics Provider.
 func NewDatadogProvider(client dynamic.Interface, mapper apimeta.RESTMapper, store Store) provider.MetricsProvider {
 	maxAge := config.Datadog.GetInt64("external_metrics_provider.local_copy_refresh_rate")
-	return &datadogProvider{
+	d := &datadogProvider{
 		client: client,
 		mapper: mapper,
-		values: make(map[provider.CustomMetricInfo]int64),
 		store:  store,
 		maxAge: maxAge,
 	}
+	go d.externalMetricsSetter()
+	return d
+}
+
+func (p *datadogProvider) externalMetricsSetter() {
+	log.Infof("Starting async loop to collect External Metrics")
+	tick := time.NewTicker(time.Duration(p.maxAge) * time.Second)
```

Comment:
Ticker needs to stop at one point
```
tick := time.NewTicker(time.Duration(p.maxAge) * time.Second)
defer  tick.Stop()
```
same for `out Timer`

Reply:
correct, thanks for pointing that out